### PR TITLE
feat(): Add copy implementation

### PR DESCRIPTION
**Pull request by 27pchrisl**
Implementation for all platforms, and uses the same copy semantics across all.

This method is called 'copy' (rather than node-style copyFile), since it can copy both directory structures and individual files.

It turns out that a copy operation has almost everything in common with a rename operation, so in each case there is a _copy method that does the bulk of the param checking/error handling etc. so as not to end up with a lot of duplication.

Copy, and now rename support a 'toDirectory' as well as the pre-existing 'directory'. Developers can omit 'directory' to use DEFAULT_DIRECTORY, include only 'directory' to rename/copy within a directory, or include both to rename/copy between directories.

Finally, added a set of tests to the example app to show the move/copy operations working.

### CODE REVIEW

**Comment by mlynch** [comment]
Looks good to me! Thanks @27pchrisl for adding this!

File: android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java [hunk 1]
```diff
@@ -455,28 +493,72 @@ public void rename(PluginCall call) {
     }
```
Comment (jcesarmobile) [inline android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java]: this is preventing from copying/renaming if the path is the same but the directory is different, you should compare `fromObject` and  `toObject` instead.

File: ios/Capacitor/Capacitor/Plugins/Filesystem.swift [hunk 2]
```diff
@@ -341,24 +341,38 @@ public class CAPFilesystemPlugin : CAPPlugin {
...
+   * Copy or rename a file or directory.
+   */
+  private func _copy(call: CAPPluginCall, doRename: Bool) {
+    guard let from = call.get("from", String.self), let to = call.get("to", String.self) else {
+      handleError(call, "Both to and from must be provided")
       return
     }
     
     let directoryOption = call.get("directory", String.self, DEFAULT_DIRECTORY)!
+    var toDirectoryOption = call.get("toDirectory", String.self, DEFAULT_DIRECTORY)!
```
[the remark this answers is not in the document]
Comment (27pchrisl) [reply ios/Capacitor/Capacitor/Plugins/Filesystem.swift]: Done

File: electron/src/electron/filesystem.ts [hunk 3]
```diff
@@ -173,24 +176,103 @@ export class FilesystemPluginElectron extends WebPlugin implements FilesystemPlu
...
+      } else {
+        copyRecursively(fromPath, toPath)
+          .then(() => resolve())
+          .catch(reject);
+      }
     });
   }
 
+  copy(options: CopyOptions): Promise<CopyResult> {
+    return this._copy(options, true);
```
[the remark this answers is not in the document]
Comment (27pchrisl) [reply electron/src/electron/filesystem.ts]: Good catch, missed this during refactoring

File: electron/src/electron/filesystem.ts [hunk 4]
```diff
@@ -173,24 +176,103 @@ export class FilesystemPluginElectron extends WebPlugin implements FilesystemPlu
...
+  }
+
+  rename(options: RenameOptions): Promise<RenameResult> {
+    return this._copy(options, false);
```
[the remark this answers is not in the document]
Comment (27pchrisl) [reply electron/src/electron/filesystem.ts]: Done

File: android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java (hunk 1 above)
Comment (27pchrisl) [reply android/capacitor/src/main/java/com/getcapacitor/plugin/Filesystem.java]: Fixed

**Comment by 27pchrisl** [comment]
I've also hugely simplified the tests into four buttons, and removed all the negative tests and kept the main gist of it.
